addons/content-index/src/main/java/org/commonjava/indy/content/index/IndexingContentManagerDecorator.java

```diff
@@ -599,6 +580,26 @@ public Transfer getTransfer( final StoreKey storeKey, final String path, final T
         return transfer;
     }
 
+    private void nfcForGroup( final ArtifactStore store, final Transfer transfer, final ConcreteResource resource )
```

Comment:
So, here's a problem we may have had before this PR:

GIVEN:

* Groups A and B
* Hosted repository X
* Both A and B contain X as a member
* Path P doesn't exist in X

WHEN :: THEN:

* Resolve P from A :: Both X and A should get NFC entries for P
* Upload P to X via B :: NFC entry for P in X should be cleared
* Resolve P from A :: **NFC entry for P in A should have been cleared, but probably hasn't**

Is there any chance we can setup a test for this, and make sure it's fixed too? I'm not sure if we'll need some sort of event listener for this, or if we should do a reverse group lookup for things containing X (and so on up the nested group structure) to find out what needs to be cleared. IIRC we can do that now in ArtifactStoreQuery / StoreDataManager.